src/pages/IstioRuleDetails/IstioRuleInfo.tsx

```diff
@@ -36,14 +36,25 @@ class IstioRuleInfo extends React.Component<RuleId, RuleInfoState> {
 
   fetchIstioRuleDetails(props: RuleId) {
     console.log('Fetching info of a service...');
-    API.GetIstioRuleDetail(props.namespace, props.rule).then(response => {
-      let data = response['data'];
-      this.setState({
-        name: data.name,
-        match: data.match,
-        actions: data.actions
+    API.GetIstioRuleDetail(props.namespace, props.rule)
+      .then(response => {
+        let data = response['data'];
+        this.setState({
+          name: data.name,
+          match: data.match,
+          actions: data.actions
+        });
+      })
+      .catch(error => {
+        let errorMessage = 'Could not fetch IstioRule details.';
```

Comment:
According to this page [1] `error` may not have a `response` nested field. Maybe we should have a shared function in API that transforms the axios error object in something more directly usable.

[1] https://github.com/axios/axios#handling-errors